File: python/meteroid/models/customer_patch_request.py
# this file is @generated
import dataclasses
import typing as t

from ..serialization import BaseModel
from .address import Address
from .currency import Currency
from .custom_tax_rate import CustomTaxRate
from .customer_type import CustomerType
from .invoicing_entity_id import InvoicingEntityId
from .shipping_address import ShippingAddress


@dataclasses.dataclass
class CustomerPatchRequest(BaseModel):
    alias: t.Optional[str] = None

    billing_address: t.Optional[Address] = None

    billing_email: t.Optional[str] = None

    currency: t.Optional[Currency] = None

    custom_properties: t.Optional[t.Any] = None
    """Partial update of custom property values (merge; send a key with `null` to remove it).
    Omit to leave unchanged."""

    custom_taxes: t.Optional[t.List[CustomTaxRate]] = None

    customer_type: t.Optional[CustomerType] = None

    exemption_reason: t.Optional[str] = None
    """Free-text legal exemption mention surfaced on exempt invoices."""

    first_name: t.Optional[str] = None

    invoicing_emails: t.Optional[t.List[str]] = None

    invoicing_entity_id: t.Optional[InvoicingEntityId] = None

    invoicing_language: t.Optional[str] = None
    """Deprecated: use `preferred_locales`. Applied only when `preferred_locales` is absent."""

    is_tax_exempt: t.Optional[bool] = None

    last_name: t.Optional[str] = None

    legal_number: t.Optional[str] = None
    """BT-47 — the buyer's national register identifier (SIREN/SIRET, HRB)."""

    name: t.Optional[str] = None

    phone: t.Optional[str] = None

    preferred_locales: t.Optional[t.List[str]] = None
    """Preferred document languages, most-preferred first (BCP-47 tags, e.g.
    `["fr-FR", "en"]`); overrides the invoicing entity default. Omit to leave
    unchanged, send `[]` to reset to that default."""

    shipping_address: t.Optional[ShippingAddress] = None

    vat_number: t.Optional[str] = None

File: go/customer_patch_request.go
// this file is @generated
package meteroid

type CustomerPatchRequest struct {
	Alias *string `json:"alias,omitempty"`

	BillingAddress *Address `json:"billing_address,omitempty"`

	BillingEmail *string `json:"billing_email,omitempty"`

	Currency *Currency `json:"currency,omitempty"`

	// Partial update of custom property values (merge; send a key with `null` to remove it).
	// Omit to leave unchanged.
	CustomProperties map[string]any `json:"custom_properties,omitempty"`

	CustomTaxes []CustomTaxRate `json:"custom_taxes,omitempty"`

	// Free-text legal exemption mention surfaced on exempt invoices.
	ExemptionReason *string `json:"exemption_reason,omitempty"`

	InvoicingEmails []string `json:"invoicing_emails,omitempty"`

	InvoicingEntityId *InvoicingEntityId `json:"invoicing_entity_id,omitempty"`

	// Preferred document language (e.g. `en-US`, `fr-FR`); overrides the invoicing entity default.
	// Omit to leave unchanged, send `""` to reset to the invoicing entity default.
	// Unsupported languages fall back to `en-US` when rendering.
	InvoicingLanguage *string `json:"invoicing_language,omitempty"`

	IsTaxExempt *bool `json:"is_tax_exempt,omitempty"`

	Name *string `json:"name,omitempty"`

	Phone *string `json:"phone,omitempty"`

	ShippingAddress *ShippingAddress `json:"shipping_address,omitempty"`

	VatNumber *string `json:"vat_number,omitempty"`
}
